game.cpp

```diff
@@ -495,6 +479,34 @@ void put_player_on_food(player &p, const food &f)
   p.place_to_position(get_position(f));
 }
 
+bool have_same_position(const player& p, const food& f)
+{
+  return p.get_x() - f.get_x() < 0.0001 &&
+      p.get_x() - f.get_x() > -0.0001 &&
+      p.get_y() - f.get_y() < 0.0001 &&
+      p.get_y() - f.get_y() > -0.0001;
+}
+
+bool in_contact_with_uneaten_food(const player &p, const food &f)
+{
+  return have_same_position(p, f) && !f.is_eaten();
+}
+
+bool any_player_food_collision(const game& g)
```

Comment:
Why not keep the naming for the 2 function functions 
```c++
has_food_collision(const game& g)
```
and 
```c++
are_colliding(const player& p, const food& f)
``` 
or 
```c++
have_collision(const player& p, const food& f)
```
? 
It is easier to read and conveys the message, especially when the test is read in #256. I find this nomenclature a bit clunky.

Reply:
Yeaaah I do have to admit the names are a bit clunky indeed. But I choose them precisely bc I feel they better convey what the function does. 

`are_colliding(p, f)` → `in_contact_with_uneaten_food(p, f)`
Not clear what is meant that "colliding" means (same position + food uneaten). From the name I thought this just meant the two objects have the same position, e.g. this function was a duplicate of `have_same_position`.
So I went with a more explicit name.

`has_food_collision(g)` → `any_player_food_collision(g)`
I prefer player-food collision over food collision only (collision with what? other food items, enemies?)
I found that "any" makes it clearer that we're iterating across all players and food items in the game, but this isn't less important.
I can see now there is an issue of consistency, I used the word collision here but not in the function it calls.

All in all I think my issue is with a "collision" not being formally defined and being a not-so-obvious description of the situation (same position + able to interact). As a naive reader, "collision" just means same position to me. Though now I can see that collision refers to two objects interacting when they meet, as opposed to merely passing through each other (and I really can't think of a better name 🤔)

Happy to revert `in_contact_with_uneaten_food(p, f)` → `are_colliding(p, f)`, but then I think a definition of "collision" in inline comments would help reading the functions. I'll do that!

Reply:
I'm reverting changes to the names:

`any_player_food_collision(g)` → `has_any_player_food_collision(g)`
`in_contact_with_uneaten_food(p, f)` -> `are_colliding(p, f)`

All in all I think "collision" is a fairly good game, but I may have something more fitting to the situation: how about
`are_colliding(p, f)` -> `can_interact(p, f)`
`has_any_player_food_collision(g)` -> `has_any_player_food_interaction(g)` ?